### PR TITLE
Added support of comparing query errors

### PR DESCRIPTION
**Pull request by mewang-tableau**

### CODE REVIEW

File: tdvt/tdvt/test_results.py [hunk 1]
```diff
@@ -98,14 +101,15 @@ def table_to_json(self):
                 for v in t.findall('value'):
                     tuple_list.append(v.text)
 
-        return {'tuples' : tuple_list}
+        return {'tuples': tuple_list}
 
     def __json__(self):
-        return {'tested_sql' : self.tested_sql, 'tested_tuples' : self.tested_tuples, 'id' : self.id, 'name' : self.name, 'sql' : self.get_sql_text(), 'table' : self.table_to_json()}
```
Comment (lukewrites) [inline tdvt/tdvt/test_results.py:104]: I know it's a hassle, but could you format the `return` statement here and in tdvtconfig.py:74 so they won't raise a flake8 error?

File: tdvt/test/tdvt_test.py [hunk 2]
```diff
@@ -55,16 +55,20 @@ def test_diff(self):
...
                 compare_tuples = True
             if 'expected.both' in test:
                 compare_tuples = True
                 compare_sql = True
+            if 'expected.error' in test:
+                compare_error = True
             test_config = TdvtInvocation()
             test_config.tested_sql = compare_sql
             test_config.tested_tuples = compare_tuples
+            test_config.tested_error = compare_error
```
[the remark this answers is not in the document]
Comment (lukewrites) [reply tdvt/test/tdvt_test.py:71]: Agree with Logan, we should get a few unit tests around these.

File: .pep8speaks.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@ scanner:
     linter: flake8
 
 flake8:
-    max-line-length: 120  # Default is 79 in PEP 8
+    max-line-length: 500  # Default is 79 in PEP 8
```
Comment (lukewrites) [inline .pep8speaks.yml:5]: Please revert to 120.

Comment (lukewrites) [reply .pep8speaks.yml:5]: For those long `return` statements, they can either be escaped or you cna add a
`#noqa: E501`
on to the end of the lines.

File: tdvt/test/tdvt_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mewang-tableau) [reply tdvt/test/tdvt_test.py:71]: Tests added in new commit.

File: tdvt/tdvt/test_results.py (hunk 1 above)
Comment (mewang-tableau) [reply tdvt/tdvt/test_results.py:104]: Formatted to complying flack8 config.

File: .pep8speaks.yml (hunk 3 above)
Comment (mewang-tableau) [reply .pep8speaks.yml]: Reverted in new commit.

**Pull request merged**